books/2015/pro-android-web-game-apps-2012/cljs/src/main/ch5/main.cljs

```diff
@@ -104,11 +104,31 @@
           x (quot (- t2 t1) 100)
           _ (async/>! >bus [:anim (fn [t] [[:knight x 200 :standing :right]])])])))
 
+(defn start-input-loop!
+  [>bus]
+  (doseq [[t k] [["keydown" :down]
+                 ["keyup" :up]]]
+    (js/document.addEventListener
+      t
+      (fn [e]
+        (when-let [m (case (.-key e)
+                       "ArrowRight" [k :right]
+                       "ArrowLeft" [k :left]
+                       nil)]
+          (async/put! >bus [:input m]))))))
+
 (defn init
   []
   (let [canvas-id "canvas"
         >bus (async/chan)
         <bus (async/pub >bus first)]
+    (let [<rcv (async/chan)]
+      (async/sub <bus :input <rcv)
```

Comment:
Why do you do a second 'let' inside your first 'let'? Any reason I don't see?

Reply:
Because this `let` is a debug-only piece of code and having it in a separate form makes it easier to remove (`da)` in Vim).